### PR TITLE
Add Legacy Master PDF Editor

### PR DESCRIPTION
**Pull request by jointri**
Had to delete my old repo and make a fresh one in order to be able to make a new PR, this one should be fixed for new PR for new package for flathub. This package is important because it is the last version to be able to make edits to a PDF without watermark. Let me know if you want any changes to this PR.

### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [ ] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [ ] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [ ] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

File: net.codeindustry.MasterPDFEditor-Legacy.yaml [hunk 1]
```diff
@@ -0,0 +1,46 @@
+app-id: net.codeindustry.MasterPDFEditor
```
Comment (hfiguiere) [inline net.codeindustry.MasterPDFEditor-Legacy.yaml:1]: also here the id should match too.

File: net.codeindustry.MasterPDFEditor-Legacy.yaml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+runtime: org.kde.Platform
+runtime-version: '5.13'
```
Comment (hfiguiere) [inline net.codeindustry.MasterPDFEditor-Legacy.yaml:3]:
```suggestion
runtime-version: '5.15-21.08'
```

5.13 is EOL

**Review by hfiguiere — CHANGES_REQUESTED** [review]

File: net.codeindustry.MasterPDFEditor4.metainfo.xml [hunk 3]
```diff
@@ -0,0 +1,36 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<component type="desktop-application">
+  <id>net.codeindustry.MasterPDFEditor4</id>
+  <launchable type="desktop-id">net.codeindustry.MasterPDFEditor.desktop</launchable>
```
Comment (hfiguiere) [inline net.codeindustry.MasterPDFEditor4.metainfo.xml]:
```suggestion
  <launchable type="desktop-id">net.codeindustry.MasterPDFEditor4.desktop</launchable>
```

Comment (hfiguiere) [reply net.codeindustry.MasterPDFEditor4.metainfo.xml]: nevermind, stream crossed.

**Comment by jointri** [comment]
Made a few changes. Let me know what you think.

File: net.codeindustry.MasterPDFEditor4.desktop [hunk 4]
```diff
@@ -0,0 +1,9 @@
+[Desktop Entry]
+Name=Master PDF Editor 4
+Comment=Edit PDF files
+Exec=masterpdfeditor %f
+Terminal=false
+Icon=net.codeindustry.MasterPDFEditor-Legacy
```
Comment (hfiguiere) [inline net.codeindustry.MasterPDFEditor4.desktop:6]:
```suggestion
Icon=net.codeindustry.MasterPDFEditor4
```

**Comment by hfiguiere** [comment]
bot, build net.codeindustry.MasterPDFEditor4

File: net.codeindustry.MasterPDFEditor4.yaml [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+    sources:
+      - type: extra-data
+        url: http://www.code-industry.net/public/
+        filename: master-pdf-editor-4.3.89_qt5.amd64.tar.gz
+        sha256: ac77db75bcc5f2bce21d98a55c07176e15682b0abe5a83708c53d38721fdab4c
+        size: 15002050
+      - type: script
+        dest-filename: apply_extra.sh
+        commands:
+          - 'tar --strip-components=1 -xf master-pdf-editor-4.3.89_qt5.amd64.tar.gzz'
```
Comment (hfiguiere) [inline net.codeindustry.MasterPDFEditor4.yaml:39]: you need to restrict the build to x86_64: https://github.com/flathub/flathub/wiki/App-Maintenance#flathubjson

(unless there is a version for aarch64)

File: net.codeindustry.MasterPDFEditor4.yaml [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+
+  - name: masterpdfeditor4
+    buildsystem: simple
+    build-commands:
+      - 'mkdir -p /app/bin'
+      - 'ln -s /app/{extra,bin}/masterpdfeditor4'
+      - 'install -Dm 755 apply_extra.sh /app/bin/apply_extra'
+      - 'install -Dm 644 net.codeindustry.MasterPDFEditor-Legacy.metainfo.xml -t /app/share/metainfo'
+      - 'install -Dm 644 net.codeindustry.MasterPDFEditor-Legacy.desktop -t /app/share/applications'
+      - 'install -Dm 644 net.codeindustry.MasterPDFEditor-Legacy.png -t /app/share/icons/hicolor/128x128/apps'
```
Comment (hfiguiere) [inline net.codeindustry.MasterPDFEditor4.yaml:29]:
```suggestion
      - 'install -Dm 644 net.codeindustry.MasterPDFEditor4.metainfo.xml -t /app/share/metainfo'
      - 'install -Dm 644 net.codeindustry.MasterPDFEditor4.desktop -t /app/share/applications'
      - 'install -Dm 644 net.codeindustry.MasterPDFEditor4.png -t /app/share/icons/hicolor/128x128/apps'
```

and you need to rename the PNG

File: net.codeindustry.MasterPDFEditor4.yaml [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+        dest-filename: apply_extra.sh
+        commands:
+          - 'tar --strip-components=1 -xf master-pdf-editor-4.3.89_qt5.amd64.tar.gzz'
+          - 'rm master-pdf-editor-4.3.89_qt5.amd64.tar.gz'
+      - type: file
+        path: net.codeindustry.MasterPDFEditor-Legacy.metainfo.xml
+      - type: file
+        path: net.codeindustry.MasterPDFEditor-Legacy.desktop
+      - type: file
+        path: net.codeindustry.MasterPDFEditor-Legacy.png
```
Comment (hfiguiere) [inline net.codeindustry.MasterPDFEditor4.yaml:46]:
```suggestion
        path: net.codeindustry.MasterPDFEditor4.metainfo.xml
      - type: file
        path: net.codeindustry.MasterPDFEditor4.desktop
      - type: file
        path: net.codeindustry.MasterPDFEditor4.png
```

**Comment by jointri** [comment]
bot, build net.codeindustry.MasterPDFEditor4

**Comment by jointri** [comment]
What does this extra data error mean?

**Comment by jointri** [comment]
bot, build net.codeindustry.MasterPDFEditor4

**Comment by jointri** [comment]
bot, build net.codeindustry.MasterPDFEditor4

**Comment by jointri** [comment]
Before anyone approves of this PR, I found the following in the build log:

INFO    src.manifest: Checking 2 external data items
INFO    src.manifest: Skipped check [1/2] archive sane-backends-1.0.28.tar.gz (from net.codeindustry.MasterPDFEditor4.yaml)
INFO    src.manifest: Started check [2/2] extra-data master-pdf-editor-4.3.89_qt5.amd64.tar.gz (from net.codeindustry.MasterPDFEditor4.yaml)
ERROR   src.manifest: Failed to check extra-data master-pdf-editor-4.3.89_qt5.amd64.tar.gz with URLChecker: Error downloading upstream source: 403, message='Forbidden', url=URL('https://www.code-industry.net:443/public/')
BROKEN: master-pdf-editor-4.3.89_qt5.amd64.tar.gz
 Couldn't get new version for http://www.code-industry.net/public/
WARNING src.main: Check finished with 1 error(s)

Should this be of any concern?

**Comment by jointri** [comment]
@hfiguiere 
What do you think of the error that I got, the build still pressed on. Was the flatpak made incorrectly?

File: net.codeindustry.MasterPDFEditor4.yaml [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+      - 'mkdir -p /app/bin'
+      - 'ln -s /app/{extra,bin}/masterpdfeditor4'
+      - 'install -Dm 755 apply_extra.sh /app/bin/apply_extra'
+      - 'install -Dm 644 net.codeindustry.MasterPDFEditor4.metainfo.xml -t /app/share/metainfo'
+      - 'install -Dm 644 net.codeindustry.MasterPDFEditor4.desktop -t /app/share/applications'
+      - 'install -Dm 644 net.codeindustry.MasterPDFEditor4.png -t /app/share/icons/hicolor/128x128/apps'
+    sources:
+      - type: extra-data
+        url: http://www.code-industry.net/public/
+        filename: master-pdf-editor-4.3.89_qt5.amd64.tar.gz
```
Comment (hfiguiere) [inline net.codeindustry.MasterPDFEditor4.yaml:33]:
```suggestion
        url: http://www.code-industry.net/public/master-pdf-editor-4.3.89_qt5.amd64.tar.gz
```

I think that's the proper one. the URL is supposed to download the file.

File: flathub.json [hunk 9]
```diff
@@ -0,0 +1,4 @@
+{
+  "skip-arches": ["aarch64"],
+  "only-arches": ["x86_64"]
```
Comment (hfiguiere) [inline flathub.json:3]:
```suggestion
  "only-arches": ["x86_64"]
```

**Comment by jointri** [comment]
bot, build net.codeindustry.MasterPDFEditor4

**Comment by jointri** [comment]
bot, build net.codeindustry.MasterPDFEditor4

**Comment by jointri** [comment]
@hfiguiere 
I think it worked out without issue. Can it be merged now?

**Comment by hfiguiere** [comment]
I can't merge. Just be patient.

**Comment by muelli** [comment]
note that the legacy version is already on flathub:
https://github.com/flathub/net.codeindustry.MasterPDFEditor/tree/4.3.89

but it may well be worth it to have the same app twice...

**Comment by hfiguiere** [comment]
1. this branch isn't published on flathub (but that can be fixed)
2. I believe that it's better to have a separate App in these circumstances (not sure everyone agree), as it is more user friendly than having app branches.

**Comment by barthalion** [comment]
```
[bpiotrowski@risuku ~]$ flatpak run net.codeindustry.MasterPDFEditor4
bwrap: execvp masterpdfeditor4: No such file or directory

[bpiotrowski@risuku ~]$ cat .local/share/flatpak/exports/share/applications/net.codeindustry.MasterPDFEditor4.desktop 
[Desktop Entry]
Name=Master PDF Editor 4
Comment=Edit PDF files
Exec=/usr/bin/flatpak run --branch=test --arch=x86_64 --command=masterpdfeditor --file-forwarding net.codeindustry.MasterPDFEditor4 @@ %f @@
Terminal=false
Icon=net.codeindustry.MasterPDFEditor4
Type=Application
Categories=Office;Graphics;
MimeType=application/pdf;application/x-bzpdf;application/x-gzpdf;
X-Flatpak=net.codeindustry.MasterPDFEditor4

[bpiotrowski@risuku ~]$ /usr/bin/flatpak run --branch=test --arch=x86_64 --command=masterpdfeditor --file-forwarding net.codeindustry.MasterPDFEditor4
bwrap: execvp masterpdfeditor: No such file or directory
```

It can be launched neither via CLI or the desktop file.

File: net.codeindustry.MasterPDFEditor4.yaml [hunk 10]
```diff
@@ -0,0 +1,46 @@
+app-id: net.codeindustry.MasterPDFEditor4
+runtime: org.kde.Platform
+runtime-version: '5.15-21.08'
+sdk: org.kde.Sdk
+command: masterpdfeditor4
```
Comment (hfiguiere) [inline net.codeindustry.MasterPDFEditor4.yaml:5]: are you sure it's the right command?

**Pull request closed without merging**